vl-proza-message.src.js

```diff
@@ -65,20 +69,50 @@ export class VlProzaMessage extends VlElement(HTMLElement) {
 
     _loadMessage() {
         if (this._domain && this._code) {
-            VlProzaMessage._getMessage(this._domain, this._code).then(message => this._contentElement.innerHTML = message.tekst);
+            this._getMessage(this._domain, this._code).then(message => this._contentElement.innerHTML = message);
         } else {
             this._contentElement.innerHTML = null;
         }
     }
 
+    _getMessage(domain, code) {
+        if (this._shouldPreload) {
+            return VlProzaMessage._getPreloadedMessage(domain, code);
+        } else {
+            return VlProzaMessage._getMessage(domain, code);
+        }
+    }
+
```

Comment:
ik zou verwachten dat de "getMessage" gewoon de juiste key in een object opzoekt en die teruggeeft en dat het al dan niet preloaden bij het "laden" van de component aangeroepen wordt. 
maar misschien persoonlijke voorkeur ...

Reply:
gaat ge trouwens op deze manier niet nog altijd hebben dat als er ene component op de pagina preload, en een andere niet, dat die 2e toch zijn eigen message nog eens gaat ophalen?

Reply:
> ik zou verwachten dat de "getMessage" gewoon de juiste key in een object opzoekt en die teruggeeft en dat het al dan niet preloaden bij het "laden" van de component aangeroepen wordt.
> maar misschien persoonlijke voorkeur ...

Door de check in _getMessage te zetten, zorgen we er ook voor dat men at runtime preloaded aan/af kan zetten. Ik kan me er niet meteen een use case voor bedenken, maar de mogelijkheid is er op deze manier wel en ik vind het zelf zo ook wel vrij duidelijk lezen.

> gaat ge trouwens op deze manier niet nog altijd hebben dat als er ene component op de pagina preload, en een andere niet, dat die 2e toch zijn eigen message nog eens gaat ophalen?

Ja, maar dat kan je niet volledig voorkomen zonder in te boeten op page load time. Stel dat je pagina aan het inladen is en de eerste ingeladen component is niet-preloaded. Die zal dan voor zijn domein en code het bericht al gaan fetchen en tonen. De enigste manier om dit te voorkomen is om te wachten tot heel de pagina geladen is vooraleer we nog maar 1 proza-message laten fetchen. Maar dat lijkt me geen goed idee.

Wat je wel kan doen is dat alle componenten de preload cache gebruiken indien die beschikbaar is (ofja, de promise), of ze nu voor preload kiezen of niet. Dat helpt wel enkel voor de messages die na de preload message gerendered worden.

Reply:
ja, ik stel mij voor dat er op de "homepage/landingpage" ne preload staat en dat al de rest die niet hebben bvb, en dan ga je toch nog alles dubbel ophalen?

Reply:
Het is natuurlijk niet omdat een webcomponent als eerste staat in de markup dat die ook als eerst zijn werk zal beginnen doen. Afwachten om te zien of er een proza-message met preload tussenzit is geen optie vanwege performance.

Dus ja, we kunnen zeggen dat een proza-message _zonder_ preload de preload cache toch gebruikt indien beschikbaar. Voor de gevallen waar een proza-message _met_ preload dan wel als (een van de) eerste in werking schiet hebben we winst. Voor andere gevallen niet, maar ook geen verlies ten opzichte van hoe het nu is.